src/www/ui/api/Controllers/UploadController.php

```diff
@@ -636,4 +637,77 @@ private function areAgentsScheduled($uploadId, $agents, $response)
     }
     return true;
   }
+
+  /**
+   * Set permissions for a upload in a folder for different groups
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setUploadPermissions($request, $response, $args)
+  {
+    $returnVal = null;
+    // checking if the scheduler is running or not
+    $commu_status = fo_communicate_with_scheduler('status', $response_from_scheduler, $error_info);
+    if ($commu_status) {
+      $response_from_scheduler = "";
+    } else {
+      $returnVal = new Info(409, "Scheduler is not running!", InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    // Initialising upload-permissions plugin
+    global $container;
+    $restHelper = $container->get('helper.restHelper');
+    $uploadPermissionObj = $restHelper->getPlugin('upload_permissions');
+
+    $dbManager = $this->dbHelper->getDbManager();
+    // parsing the request body
+    $reqBody = $this->getParsedBody($request);
+
+    $folder_pk = intval($reqBody['folderId']);
+    $upload_pk = intval($args['id']);
+    $upload = $this->uploadAccessible($this->restHelper->getGroupId(), $upload_pk);
+    if ($upload !== true) {
+      return $response->withJson($upload->getArray(), $upload->getCode());
+    }
+    $allUploadsPerm = ($reqBody['allUploadsPermission'] == 'yes') ? 1 : 0;
+    $perm_upload_pk = intval($reqBody['permupk']);
+    $perm = intval($reqBody['perm']);
```

Comment:
`permupk` and `perm` are not sent from the client.

Reply:
Yes, I'll remove it!